src/transaction/coinselector-aggregate.ts

```diff
@@ -141,26 +160,79 @@ export class AggregateCoinSelector extends AbstractCoinSelector {
     }
 
     // gets the input UTXOs to cover the fee
-    private getInputsForFee(inputsToAggregate: UTXO[], eligibleUTXOs: UTXO[], fee: bigint): UTXO[] {
-        // get the input UTXOs that are not included in inputsToAggregate
-        const eligiblefeeInputs = eligibleUTXOs.filter(
-            (utxo) => !inputsToAggregate.some((input) => input.txhash === utxo.txhash && input.index === utxo.index),
-        );
+    private getInputsForFee(
+        inputsToAggregate: UTXO[],
+        availableUTXOs: UTXO[],
+        fee: bigint,
+    ): { feeInputs: UTXO[]; updatedInputsToAggregate: UTXO[] } {
+        let currentInputsToAggregate = [...inputsToAggregate];
+
+        while (currentInputsToAggregate.length > 0) {
+            // get the input UTXOs that are not included in currentInputsToAggregate
+            const eligiblefeeInputs = availableUTXOs.filter(
+                (utxo) =>
+                    !currentInputsToAggregate.some(
+                        (input) => input.txhash === utxo.txhash && input.index === utxo.index,
+                    ),
+            );
+
+            // If no eligible fee inputs available, we need to remove inputs from aggregation
+            if (eligiblefeeInputs.length === 0) {
+                console.warn(
+                    `No UTXOs available for fee coverage, removing minimum UTXOs from aggregation to cover fee`,
+                );
+
+                // Sort by denomination ascending to remove the smallest first
+                const sortedInputsToAggregate = this.sortUTXOsByDenomination(currentInputsToAggregate, 'asc');
+
+                // Calculate minimum number of UTXOs to remove to cover the fee
+                let feeSum = BigInt(0);
+                let utxosToRemove = 0;
+                for (const utxo of sortedInputsToAggregate) {
+                    feeSum += BigInt(denominations[utxo.denomination!]);
+                    utxosToRemove++;
+                    if (feeSum >= fee) {
```

Comment:
doesn't the fee technically change with change of inputs and outputs during to fee calculation? curious if there are edge cases